nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-mongodb/src/main/java/org/nuxeo/mongodb/audit/MongoDBAuditBackend.java

```diff
@@ -129,29 +125,57 @@ public void onApplicationStopped() {
+        // prepare parameters
+        List<AuditFilterBuilder> filters = builder.filters();
+        List<AuditOrderBuilder> orders = builder.orders();
+        long offset = builder.offset();
+        long limit = builder.limit();
+
+        // create MongoDB filter
+        List<Bson> filterList = new ArrayList<>(filters.size());
+        for (AuditFilterBuilder filter : filters) {
+            switch (filter.getOperator()) {
+            case AuditFilterBuilder.OPERATOR_LT:
+                filterList.add(Filters.lt(filter.getName(), filter.getValue()));
+                break;
+            case AuditFilterBuilder.OPERATOR_LTE:
+                filterList.add(Filters.lte(filter.getName(), filter.getValue()));
+                break;
+            case AuditFilterBuilder.OPERATOR_GTE:
+                filterList.add(Filters.gte(filter.getName(), filter.getValue()));
+                break;
+            case AuditFilterBuilder.OPERATOR_GT:
+                filterList.add(Filters.gt(filter.getName(), filter.getValue()));
+                break;
+            case AuditFilterBuilder.OPERATOR_IN:
+                filterList.add(Filters.in(filter.getName(), (List<?>) filter.getValue()));
+                break;
+            case AuditFilterBuilder.OPERATOR_EQ:
+            default:
+                filterList.add(Filters.eq(filter.getName(), filter.getValue()));
+                break;
+            }
+        }
+        Bson mgFilter = Filters.and(filterList);
 
-    @Override
-    public List<LogEntry> getLogEntriesFor(String uuid, Map<String, FilterMapEntry> filterMap, boolean doDefaultSort) {
-        Bson docFilter = Filters.eq(PROPERTY_DOC_UUID, uuid);
-
-        Bson filter;
-        if (MapUtils.isEmpty(filterMap)) {
-            filter = docFilter;
-        } else {
-            List<Bson> list = new ArrayList<>(Collections.singleton(docFilter));
-            for (Entry<String, FilterMapEntry> entry : filterMap.entrySet()) {
-                FilterMapEntry filterEntry = entry.getValue();
-                list.add(Filters.eq(filterEntry.getColumnName(), filterEntry.getObject()));
+        // create MongoDB order
+        List<Bson> orderList = new ArrayList<>(orders.size());
+        for (AuditOrderBuilder order : orders) {
+            if (order.isDescending()) {
+                orderList.add(Sorts.descending(order.getName()));
+            } else {
+                orderList.add(Sorts.ascending(order.getName()));
             }
-            filter = Filters.and(list);
         }
-        return getLogEntries(filter, doDefaultSort);
+        Bson mgOrder = Sorts.orderBy(orderList);
+
+        logRequest(mgFilter, mgOrder);
+        FindIterable<Document> iterable = collection.find(mgFilter)
+                                                    .sort(mgOrder)
+                                                    .skip(Long.valueOf(offset).intValue())
+                                                    .limit(Long.valueOf(limit).intValue());
```

Comment:
`Long.valueOf(offset).intValue()` is strange given that `offset` is already a `long`. Just cast to `int`.